contracts/lockdrop/src/migration.rs

```diff
@@ -17,4 +20,55 @@ pub struct PoolInfoV101 {
     pub is_staked: bool,
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct PoolInfoV111 {
+    pub terraswap_pool: Addr,
+    pub terraswap_amount_in_lockups: Uint128,
+    pub migration_info: Option<MigrationInfo>,
+    /// Share of total ASTRO incentives allocated to this pool
+    pub incentives_share: u64,
+    /// Weighted LP Token balance used to calculate ASTRO rewards a particular user can claim
+    pub weighted_amount: Uint256,
+    /// Ratio of Generator ASTRO rewards accured to astroport pool share
+    pub generator_astro_per_share: Decimal,
+    /// Ratio of Generator Proxy rewards accured to astroport pool share
+    pub generator_proxy_per_share: Decimal,
+    /// Boolean value indicating if the LP Tokens are staked with the Generator contract or not
+    pub is_staked: bool,
+    /// Flag defines whether the asset has rewards or not
+    pub has_asset_rewards: bool,
+}
+
 pub const ASSET_POOLS_V101: Map<&Addr, PoolInfoV101> = Map::new("LiquidityPools");
+pub const ASSET_POOLS_V111: Map<&Addr, PoolInfoV111> = Map::new("LiquidityPools");
+
+pub fn migrate_generator_proxy_per_share_to_v120(
+    deps: &DepsMut,
+    generator_proxy_per_share_old: Decimal,
+    generator: &Addr,
+    migration_info: Option<MigrationInfo>,
+) -> StdResult<RestrictedVector<AssetInfo, Decimal>> {
+    let mut generator_proxy_per_share = RestrictedVector::default();
+    if !generator_proxy_per_share_old.is_zero() {
+        let reward_info: RewardInfoResponse = deps.querier.query_wasm_smart(
+            generator,
+            &GenQueryMsg::RewardInfo {
+                lp_token: migration_info
+                    .expect("Should be migrated!")
```

Comment:
If any generator was not migrated then the whole migration would fail. Is it correct?

Reply:
It is migrated because it has already generator_proxy_per_share in the storage